File: android/KTPlayConversion.java
package org.cocos2dx.lua;

import com.ktplay.open.KTLeaderboardPaginator;
import com.ktplay.open.KTRewardItem;
import com.ktplay.open.KTUser;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.json.JSONStringer;

import java.util.ArrayList;
import java.util.jar.Pack200;


/**
 * Created by HelloMeowLab on 5/8/16.
 */
public class KTPlayConversion {

    public static JSONObject jsonWithKTUser(KTUser user) {
        JSONObject o = new JSONObject();
        try {
            o.put("userId", user.getUserId());
            o.put("headerUrl", user.getHeaderUrl());
            o.put("nickname", user.getNickname());
            o.put("gender", user.getGender());
            o.put("city", user.getCity());
            o.put("score", user.getScore());
            o.put("birthday", user.getBirthday());
            o.put("rank", user.getRank());
            o.put("scoreTag", user.getScoreTag());
            o.put("originScore", user.getOriginScore());
            o.put("loginType", user.getLoginType());
            o.put("gameUserId", user.getGameUserId());
            o.put("needPresentNickname", user.getNeedPresentNickname());
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return o;
    }

    public static JSONStringer jsonWithKTLeaderboardPaginator(KTLeaderboardPaginator leaderboard) {
        JSONStringer o = new JSONStringer();

        try {
            o.object();

            // add properties
            o.key("total");o.value(leaderboard.getTotal());
            o.key("nextCursor");o.value(leaderboard.getNextCursor());
            o.key("previousCursor");o.value(leaderboard.getPreviousCursor());
            o.key("itemCount");o.value(leaderboard.getItemCount());
            o.key("leaderboardName");o.value(leaderboard.getLeaderboardName());
            o.key("leaderboardIcon");o.value(leaderboard.getLeaderboardIcon());
            o.key("leaderboardId");o.value(leaderboard.getLeaderboardId());
            o.key("myRank");o.value(leaderboard.getMyRank());
            o.key("myScore");o.value(leaderboard.getMyScore());
            o.key("myScoreTag");o.value(leaderboard.getMyScoreTag());
            o.key("myOriginScore");o.value(leaderboard.getMyOriginScore());
            o.key("periodicalSummaryId");o.value(leaderboard.getPeriodicalSummaryId());

            // add users
            JSONArray a = new JSONArray();
            for (KTUser user : leaderboard.getUsers()) {
                a.put(jsonWithKTUser(user));
            }
            o.key("items");o.value(a);

            o.endObject();
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return o;
    }

    public static JSONStringer jsonWithKTRewardArray(ArrayList<KTRewardItem> rewards) {
        JSONStringer o = new JSONStringer();

        try {
            o.object();

            for(KTRewardItem reward : rewards) {
                o.key(reward.getTypeId());
                o.value(reward.getValue());
            }

            o.endObject();

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return o;
    }
}

File: ios/KTPlayConversion.m
//
//  KTPlayConversion.m
//
//  Created by HelloMeowLab on 5/8/16.
//
//

#import "KTPlayConversion.h"
#import "KTUser.h"
#import "KTLeaderboard.h"
@implementation KTPlayConversion

static void safe_set_dict(NSMutableDictionary* dict, NSString* key, id value)
{
    if (value) {
        [dict setObject:value forKey:key];
    } else {
        NSLog(@"value for key[%@] is nil", key);
    }
}

+ (NSDictionary*) dictionaryWithKTUser:(KTUser*)user
{
    id dict = [NSMutableDictionary dictionary];

    safe_set_dict(dict, @"userId",              user.userId);
    safe_set_dict(dict, @"headerUrl",           user.headerUrl);
    safe_set_dict(dict, @"nickname",            user.nickname);
    safe_set_dict(dict, @"gender",              @(user.gender));
    safe_set_dict(dict, @"city",                user.city);
    safe_set_dict(dict, @"score",               user.score);
    safe_set_dict(dict, @"rank",                @(user.rank));
    safe_set_dict(dict, @"snsUserId",           user.snsUserId);
    safe_set_dict(dict, @"loginType",           user.loginType);
    safe_set_dict(dict, @"gameUserId",          user.gameUserId);
    safe_set_dict(dict, @"needPresentNickname", @(user.needPresentNickname));
    safe_set_dict(dict, @"originScore",         @(user.originScore));
    safe_set_dict(dict, @"scoreTag",            user.scoreTag);

    return dict;
}


+ (NSDictionary*) dictionaryWithKTLeaderboardPaginator:(KTLeaderboardPaginator*)leaderboard
{
    id dict = [NSMutableDictionary dictionary];

    safe_set_dict(dict, @"total",               @(leaderboard.total));
    safe_set_dict(dict, @"nextCursor",          leaderboard.nextCursor);
    safe_set_dict(dict, @"previousCursor",      leaderboard.previousCursor);
    safe_set_dict(dict, @"leaderboardName",     leaderboard.leaderboardName);
    safe_set_dict(dict, @"leaderboardIcon",     leaderboard.leaderboardIcon);
    safe_set_dict(dict, @"leaderboardId",       leaderboard.leaderboardId);
    safe_set_dict(dict, @"myRank",              @(leaderboard.myRank));
    safe_set_dict(dict, @"myScore",             leaderboard.myScore ? leaderboard.myScore : @"0");
    safe_set_dict(dict, @"myOriginScore",       @(leaderboard.myOriginScore));
    safe_set_dict(dict, @"myScoreTag",          leaderboard.myScoreTag ? leaderboard.myScoreTag : @"default");

    id users = [NSMutableArray arrayWithCapacity:leaderboard.items.count];
    for (KTUser* user in leaderboard.items) {
        [users addObject:[KTPlayConversion dictionaryWithKTUser:user]];
    }
    [dict setObject:users forKey:@"items"];

    return dict;
}

+ (NSDictionary*) dictionaryWithKTRewards:(NSArray*)rewards
{
    id dict = [NSMutableDictionary dictionary];

    for (KTRewardItem *item in rewards) {
        [dict setObject:@(item.value) forKey:item.typeId];
    }

    return dict;
}

@end
